nx/source/services/audout.c

```diff
@@ -0,0 +1,387 @@
+#include <string.h>
+#include "types.h"
+#include "result.h"
+#include "ipc.h"
+#include "services/audout.h"
+#include "services/sm.h"
+
+#define DEVICE_NAME_LENGTH 0x100
+#define DEFAULT_SAMPLE_RATE 0xBB80
+#define CHANNEL_COUNT_MAGIC 0xCAFE0000
+
+static Service g_audoutSrv;
+static Service g_audoutIAudioOut;
+
+Result audoutInitialize(void)
+{
+    if (serviceIsActive(&g_audoutSrv))
+        return MAKERESULT(Module_Libnx, LibnxError_AlreadyInitialized);
+
+    Result rc = 0;
+    rc = smGetService(&g_audoutSrv, "audout:u");
+    
+    // Setup the default device
+    if (R_SUCCEEDED(rc))
+    {
+        u32 SampleRate = 0;
+        u32 ChannelCount = 0;
+        PcmFormat Format = 0;
+        AudioOutState State = 0;
+        
+        // Passing an empty device name will open the default "DeviceOut"
+        char DeviceNameIn[DEVICE_NAME_LENGTH];
```

Comment:
This is uninitialized... perhaps it should have ` = {0}` at the end?

Reply:
Indeed it should